### PR TITLE
Hide the deploy stage if we are not on master

### PR DESCRIPTION
**Pull request by AustinShalit**
## Description

This will hide the deploy stage if we are not building master.

## Motivation and Context

Improves PR build time by ~80 seconds

## How Has This Been Tested?

pr: https://travis-ci.org/AustinShalit/travis-test/builds/365405661
push to branch: https://travis-ci.org/AustinShalit/travis-test/builds/365405602
push to master: https://travis-ci.org/AustinShalit/travis-test/builds/365405877

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change API specifications or require data migrations)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -32,6 +32,7 @@ before_deploy:
 jobs:
   include:
   - stage: deploy
+    if: branch = master
```
Comment (bdaroz) [inline .travis.yml]: I guess CI will confirm in a few mins but, if you want this to skip if the branch isn't master, is this the right equality?

Comment (AustinShalit) [reply .travis.yml]: I think so.  I am following the example here: https://docs.travis-ci.com/user/build-stages/#Specifying-Stage-Order-and-Conditions

Comment (AustinShalit) [reply .travis.yml]: Oops... looks like I put it in the wrong spot.  Will fix.

**Comment by phil-lopreiato** [comment]
This looked like it still scheduled a deploy job, just that it now runs first?

**Comment by AustinShalit** [comment]
Hmm... looks like I missed the part of the documentation: "for pull requests: the base branch name".  Going to try again

**Comment by AustinShalit** [comment]
This looks ok to me now.  Is the example build from my travis test repo:
pr: https://travis-ci.org/AustinShalit/travis-test/builds/365405661
push to branch: https://travis-ci.org/AustinShalit/travis-test/builds/365405602
push to master: https://travis-ci.org/AustinShalit/travis-test/builds/365405877

**Pull request merged**